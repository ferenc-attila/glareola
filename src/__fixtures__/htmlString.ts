export const htmlString: string = `<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-2">
</head>
<body>
<div class="Keret">
    <div class="TonusosDoboz">
        <div class="Kozep">
            <table border="0" cellpadding="6" cellspacing="0" width="100%">
                <tr class="formtable">
                    <td width="1">&nbsp;</td>
                    <td colspan="2">
                        <table border="0" cellpadding="3" cellspacing="0" class="formtable" width="100%">
                            <tr>
                                <td width="125"><strong>Időpont:</strong></td>
                                <td>1935. december 28.</td>
                                <td rowspan="8" valign="top" align="right" width="16">&nbsp;</td>
                            </tr>
                            <tr>
                                <td width="125"><strong>Faj:</strong></td>
                                <td><a href="javascript:loginAlert()" class="Bordo">Mezei veréb</a> - <em>Passer
                                    montanus</em></td>
                            </tr>
                            <tr>
                                <td width="125"><strong>Példány, kor, ivar:</strong></td>
                                <td>1,&nbsp;&nbsp;ad.</td>
                            </tr>
                            <tr>
                                <td width="125"><strong>Település, megye:</strong></td>
                                <td><span class="text">
									Eger, Heves								</span></td>
                            </tr>

                            <tr>
                                <td width="125"><strong>Terület:</strong></td>
                                <td>Nagy-Eged</td>
                            </tr>
                            <tr>
                                <td width="125"><strong>Megfigyelő(k):</strong></td>
                                <td>John Doe, Jane Doe</td>
                            </tr>
                            <tr>
                                <td width="125"><strong>Feltöltő:</strong></td>
                                <td><a class="Bordo" href="http://birding.hu/useradatlap/0000">Jack Doe</a></td>
                            </tr>

                            <tr>
                                <td width="125"><strong>Megjegyzés:</strong></td>
                                <td>
                                    <div style="word-wrap:break-word;width:500px">Testing data</div>
                                </td>
                            </tr>

                            <tr>
                                <script type="text/javascript">

                                    var gMapLat = '47.927998';
                                    var gMapLong = '20.410659';


                                    function callGoogleMaps(lat, long) {
                                        window.open('https://www.google.com/maps/place/' + lat + 'N,' + long + 'E', '2_blank');
                                    }

                                    function showAddress(address) {
                                        console.log(address);
                                        a = address.replace(/, Magyaror.*/g, '');
                                        console.log(a);
                                        $.getJSON('https://dev.utcakereso.hu/terkep.php/geocoder/get/' + a, function (data) {
                                                lon = parseFloat(data.lon).toFixed(6);
                                                lat = parseFloat(data.lat).toFixed(6);

                                                if (isNaN(lon)) lon = 19.56;
                                                if (isNaN(lat)) lat = 47.16;

                                                $('#longitude').val(lat);
                                                $('#lattitude').val(lon);

                                                gMapLat = lat;
                                                gMapLong = lon;

                                                newLatLng = new L.LatLng(lat, lon);
                                                marker.setLatLng(newLatLng).bindTooltip(lon + ',' + lat);
                                                ;
                                                mymap.setView(newLatLng);
                                            }
                                        );
                                    }

                                    $('document').ready(function () {
                                    });
                                </script>


                                <td class="szoveg">
                                    <strong>GPS-koordináták: </strong>
                                </td>
                                <td>
                                    <span style="white-space:nowrap;display:inline;">É:&nbsp;<input type="text"
                                                                                                    name="longitude"
                                                                                                    id="longitude"
                                                                                                    readonly
                                                                                                    style="width:80px;"
                                                                                                    value="47.927998"></span>
                                    <span style="white-space:nowrap;display:inline;">K:&nbsp;<input type="text"
                                                                                                    name="lattitude"
                                                                                                    id="lattitude"
                                                                                                    readonly
                                                                                                    style="width:80px;"
                                                                                                    value="20.410659"></span>
                                    <input type="button" value="google"
                                           onclick="javascript: callGoogleMaps(gMapLat, gMapLong);">
                                </td>
                            </tr>
                            <tr>
                                <td>&nbsp;</td>
                                <td>&nbsp;</td>
                            </tr>
                            <tr>
                                <td colspan="2">
                                    <div id="map_canvas" style="z-index:0;width:395px;height:220px;"></div>
                                    <link rel="stylesheet" href="https://unpkg.com/leaflet@1.4.0/dist/leaflet.css"
                                          integrity="sha512-puBpdR0798OZvTTbP4A8Ix/l+A4dHDD0DGqYW6RQ+9jxkRFclaxxQb/SJAWZfWAkuyeQUytO7+7N4QKrDh+drA=="
                                          crossorigin=""/>
                                    <script src="https://unpkg.com/leaflet@1.4.0/dist/leaflet.js"
                                            integrity="sha512-QVftwZFqvtRNi0ZyCtsznlKSWOStnDORoefr1enyq5mVL4tmKB3S/EnC3rRJcxCPavG10IcrVGSmPh6Qw5lwrg=="
                                            crossorigin=""></script>

                                    <script>
                                        var mymap = L.map('map_canvas').setView([47.927998, 20.410659], 13);
                                        L.tileLayer('https://utcakereso.hu/tile/osm/{z}/{x}/{y}.png?access_token={accessToken}', {
                                                attribution: 'Map data &copy; <a href="https://www.openstreetmap.org/">OpenStreetMap</a>, <a href="https://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>,<a href="https://utcakereso.hu/">Utcakereso.hu</a>',
                                                maxZoom: 18,
                                                id: 'mapbox.streets',
                                                accessToken: 'your.mapbox.access.token'
                                            }
                                        ).addTo(mymap);


                                        var marker = L.marker([47.927998, 20.410659],
                                            {
                                                draggable: false
                                            }
                                        ).addTo(mymap);
                                    </script>

                                </td>
                                <td width="240" valign="top" align="center"><a class="thickbox" href="../../assets/images/glareola_pratincola.png"><img src="../../assets/images/glareola_pratincola.png" border="0"></a>						<br>
							<a href="http://www.birding.hu/kepertekeles/00000" class="Bordo">Értékelés:</a> 5/5

						 <a href="http://www.birding.hu/kepertekeles/00000" class="Bordo">Kommentek száma:</a> 0
					</td>
                            </tr>

                        </table>
                        <br>
                    </td>

                </tr>
            </table>
            <script type="text/javascript">
                function loginAlert() {
                    alert('A tartalom megtekintéséhez be kell jelentkezni!');
                }
            </script>
        </div>
    </div>
</div>
</body>
</html>`;
