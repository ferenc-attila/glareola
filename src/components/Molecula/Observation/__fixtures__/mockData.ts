import { SOURCES } from '../../../../constants';
import { IBirdingHuData } from '../../../../types/interfaces';

export const mockBirdingHuData: IBirdingHuData = {
    source: SOURCES.BIRDING_HU,
    webId: 1,
    speciesHun: 'Székicsér',
    speciesSci: 'Glareola pratincola',
    individuals: 1,
    date: '2021-05-01',
    locality: 'Budapest',
    county: 'Budapest',
    area: 'Parlament',
    latitude: 47.497913,
    longitude: 19.040236,
    observers: ['John Doe', 'Jane Doe'],
    uploader: 'John Doe',
    notes: 'Some comment',
    imageLink: 'assets/images/glareola_pratincola.png',
};
