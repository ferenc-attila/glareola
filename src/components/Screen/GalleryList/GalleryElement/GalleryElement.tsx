import React, { useState } from 'react';
import { Image, Modal, StyleSheet, Text, View } from 'react-native';

import { ICON_NAMES } from '../../../../constants';
import i18n from '../../../../localization';
import { COLORS, CONTAINER_STYLES, SPACING, TEXT_STYLES } from '../../../../styles';
import { IBirdingHuGalleryElement } from '../../../../types/interfaces';
import { GlareolaButton } from '../../../Atomic/GlareolaButton';

export const GalleryElement = (data: IBirdingHuGalleryElement) => {
    const [modalVisible, setModalVisible] = useState(false);
    return (
        <>
            <View style={styles.galleryContainer}>
                <Text style={styles.sectionTitleText}>{data.speciesHun}</Text>
                <View>
                    <Image
                        style={styles.listImage}
                        source={{ uri: data.smallImageLink }}
                        accessibilityRole='image'
                        accessibilityLabel={`${i18n.t('GALLERY_INDEX_IMAGE_DESCRIPTION')} ${data.speciesSci}`}
                    />
                </View>
                {data.locality && <Text style={styles.firstCommonText}>{data.locality}</Text>}
                {data.area && <Text style={styles.commonText}>{data.area}</Text>}
                <Text style={styles.commonText}>{data.date}</Text>
                {data.notes && <Text style={styles.commonText}>{data.notes}</Text>}
                {data.cameraType && <Text style={styles.commonText}>{data.cameraType}</Text>}
                {data.cameraSettings && data.cameraSettings.some(setting => setting !== '') && (
                    <Text style={styles.commonText}>{data.cameraSettings.join(', ')}</Text>
                )}
                <Text style={styles.lastCommonText}>{data.author}</Text>
                <View style={styles.galleryButtonContainer}>
                    <GlareolaButton
                        iconName={ICON_NAMES.IMAGE}
                        onPress={() => {
                            setModalVisible(!modalVisible);
                        }}
                        accessibilityLabel={i18n.t('BUTTON_IMAGE_LABEL')}
                    />
                </View>
            </View>
            <Modal style={{ padding: SPACING.LARGE }} animationType='fade' transparent visible={modalVisible}>
                <View style={styles.modal}>
                    <Image
                        style={styles.image}
                        source={{ uri: data.imageLink }}
                        accessibilityRole='image'
                        accessibilityLabel={`${i18n.t('GALLERY_IMAGE_DESCRIPTION')} ${data.speciesSci}`}
                    />
                    <GlareolaButton
                        onPress={() => setModalVisible(!modalVisible)}
                        iconName={ICON_NAMES.BACK}
                        accessibilityLabel={i18n.t('BUTTON_CLOSE_LABEL')}
                    />
                </View>
            </Modal>
        </>
    );
};

const styles = StyleSheet.create({
    galleryContainer: {
        ...CONTAINER_STYLES.COMMON_CONTAINER,
    },
    galleryButtonContainer: {
        ...CONTAINER_STYLES.BUTTON_CONTAINER,
    },
    modal: {
        backgroundColor: COLORS.BACKGROUND.DATA,
        padding: SPACING.LARGE,
        marginTop: SPACING.LARGE,
        width: '90%',
        height: '90%',
        alignItems: 'center',
        justifyContent: 'space-around',
        alignSelf: 'center',
        borderRadius: SPACING.SMALL,
    },
    listImage: {
        width: 150,
        height: 75,
        resizeMode: 'contain',
    },
    image: {
        margin: SPACING.MEDIUM,
        width: '100%',
        height: '90%',
        resizeMode: 'contain',
    },
    sectionTitleText: {
        ...TEXT_STYLES.SECTION_TITLE,
    },
    commonText: {
        ...TEXT_STYLES.COMMON,
    },
    firstCommonText: {
        ...TEXT_STYLES.COMMON,
        marginTop: SPACING.MEDIUM,
    },
    lastCommonText: {
        ...TEXT_STYLES.COMMON,
        marginBottom: SPACING.MEDIUM,
    },
});
