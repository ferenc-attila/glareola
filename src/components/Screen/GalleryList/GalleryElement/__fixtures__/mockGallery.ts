import { SOURCES } from '../../../../../constants';
import { IBirdingHuGalleryElement } from '../../../../../types/interfaces';

export const mockBirdingHuGalleryElement: IBirdingHuGalleryElement = {
    id: '12345678',
    author: 'John Doe',
    source: SOURCES.BIRDING_HU,
    speciesHun: 'Székicsér',
    speciesSci: 'Glareola pratincola',
    date: '2021-05-01',
    locality: 'Budapest',
    area: 'Parlament',
    notes: 'Some comment',
    smallImageLink: '../src/assets/images/glareola_icon.png',
    detailsUrl: '../../__fixtures__/birdingHuHtmlStrings.ts',
    cameraType: 'Canon EOS 5D Mark IV',
    cameraSettings: ['f/5.6', '1/1000', 'ISO 400'],
    imageLink: '../src/assets/images/glareola_pratincola.png',
};
