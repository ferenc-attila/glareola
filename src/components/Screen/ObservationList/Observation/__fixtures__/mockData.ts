import { SOURCES } from '../../../../../constants';
import { IBirdingHuData } from '../../../../../types/interfaces';

export const mockBirdingHuData: IBirdingHuData = {
    source: SOURCES.BIRDING_HU,
    details: 'http://birding.hu/megfigyeles/adatlap/12345678',
    webId: 1,
    speciesHun: 'Székicsér',
    speciesSci: 'Glareola pratincola',
    individuals: 1,
    date: '2021-05-01',
    locality: 'Budapest',
    county: 'Budapest',
    area: 'Parlament',
    latitude: 47.497913,
    longitude: 19.040236,
    observers: ['John Doe', 'Jane Doe'],
    uploader: 'John Doe',
    notes: 'Some comment',
    imageLink: '../src/assets/images/glareola_pratincola.png',
};
