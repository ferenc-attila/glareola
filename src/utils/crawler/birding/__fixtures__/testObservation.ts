import { IBirdingHuData } from '../../../../types/interfaces';

export const testObservation: IBirdingHuData = {
    source: 'birding.hu',
    webId: 12345678,
    date: '1935. december 28.',
    speciesHun: 'Mezei ver√©b',
    speciesSci: 'Passer montanus',
    individuals: 1,
    locality: 'Eger',
    county: 'Heves',
    area: 'Nagy-Eged',
    observers: ['John Doe', 'Jane Doe'],
    uploader: 'Jack Doe',
    notes: 'Testing data',
    latitude: 47.927998,
    longitude: 20.410659,
    imageLink: '../src/assets/images/glareola_pratincola.png',
};
