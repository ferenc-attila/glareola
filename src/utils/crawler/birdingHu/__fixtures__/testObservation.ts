import { SOURCES } from '../../../../constants';
import { IBirdingHuData } from '../../../../types/interfaces';

export const testObservation: IBirdingHuData = {
    source: SOURCES.BIRDING_HU,
    details: 'http://birding.hu/megfigyeles/adatlap/12345678',
    webId: 12345678,
    date: '1935. december 28.',
    speciesHun: 'Mezei veréb',
    speciesSci: 'Passer montanus',
    individuals: 1,
    locality: 'Eger',
    county: 'Heves',
    area: 'Nagy-Eged',
    observers: ['John Doe', 'Jane Doe'],
    uploader: 'Jack Doe',
    notes: 'Testing data',
    latitude: 47.927998,
    longitude: 20.410659,
    imageLink: '../src/assets/images/glareola_pratincola.png',
};

export const testGalleryElement = {
    id: '12345678',
    author: 'John Doe',
    source: SOURCES.BIRDING_HU,
    speciesHun: 'Székicsér',
    speciesSci: 'Glareola pratincola',
    date: '2021-05-01',
    locality: 'Budapest',
    area: 'Parlament',
    notes: 'Some comment',
    smallImageLink: '../src/assets/images/glareola_icon.png',
    detailsUrl: 'http://birding.hu/megfigyeles/adatlap/12345678',
    cameraType: 'Canon EOS 5D Mark IV',
    cameraSettings: ['f/5.6', '1/1000', 'ISO 400'],
    imageLink: '../src/assets/images/glareola_pratincola.png',
};
