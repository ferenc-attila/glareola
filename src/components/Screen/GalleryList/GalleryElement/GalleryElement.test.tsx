import { fireEvent, render } from '@testing-library/react-native/build/pure';
import { screen } from '@testing-library/react-native/build/screen';

import { GalleryElement } from './GalleryElement';
import { mockBirdingHuGalleryElement } from './__fixtures__/mockGallery';

describe('GalleryElement', () => {
    test('should render correctly', () => {
        const container = render(<GalleryElement {...mockBirdingHuGalleryElement} />);
        expect(container.toJSON()).toMatchSnapshot();
    });

    test('should render image button', () => {
        render(<GalleryElement {...mockBirdingHuGalleryElement} />);
        expect(screen.getByLabelText('Show image button')).toBeTruthy();
    });

    test('should open image modal if image button pressed', async () => {
        render(<GalleryElement {...mockBirdingHuGalleryElement} />);
        const imageButton = screen.getByLabelText('Show image button');
        fireEvent.press(imageButton);
        const image = screen.getByLabelText('Larger image Glareola pratincola');
        expect(image.props.accessibilityRole).toEqual('image');
        expect(image.props.source.uri).toEqual(mockBirdingHuGalleryElement.imageLink);
    });
});
