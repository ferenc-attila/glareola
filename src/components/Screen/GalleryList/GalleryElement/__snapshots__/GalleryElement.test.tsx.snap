// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryElement should render correctly 1`] = `
[
  <View
    style={
      {
        "alignContent": "center",
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#595947",
        "borderRadius": 10,
        "margin": 10,
        "padding": 10,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "700",
          "padding": 10,
          "textAlign": "center",
          "textAlignVertical": "bottom",
        }
      }
    >
      Székicsér
    </Text>
    <View>
      <Image
        accessibilityLabel="Index image Glareola pratincola"
        accessibilityRole="image"
        source={
          {
            "uri": "../src/assets/images/glareola_icon.png",
          }
        }
        style={
          {
            "height": 75,
            "resizeMode": "contain",
            "width": 150,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
          "marginTop": 10,
        }
      }
    >
      Budapest
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Parlament
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      2021-05-01
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Some comment
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Canon EOS 5D Mark IV
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      f/5.6, 1/1000, ISO 400
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "500",
          "marginBottom": 10,
        }
      }
    >
      John Doe
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#4d9460",
            "borderRadius": 5,
            "color": "#fff",
            "margin": 5,
            "padding": 10,
          }
        }
      >
        <View
          accessibilityLabel="Show image button"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text />
        </View>
      </View>
    </View>
  </View>,
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    style={
      {
        "padding": 15,
      }
    }
    transparent={true}
    visible={false}
  />,
]
`;
